packages/@aws-cdk/aws-amplify/lib/app.ts

```diff
@@ -0,0 +1,347 @@
+   */
+  public addBranch(id: string, props: BranchBaseProps): Branch {
+    const branchProps = {
+      app: this,
+      ...props
+    };
+
+    return new Branch(this, id, branchProps);
+  }
+}
+
+/**
+ * App Interface
+ */
+export interface IApp extends IResource {
+  /**
+   * App ID
+   *
+   * @attribute
+   */
+  readonly appId: string;
+}
+
+/**
+ * App Properties
+ */
+export interface AppProps {
+  /**
+   * Personal Access token for 3rd party source control system for an Amplify App, used to create webhook
+   * and read-only deploy key. Token is not stored.
+   *
+   * @default - No access token
+   */
+  readonly accessToken?: string;
+
+  /**
+   * Credentials for Basic Authorization for an Amplify App.
+   *
+   * @default - No basic auth config
+   */
+  readonly basicAuthConfig?: BasicAuthConfig;
+
+  /**
+   * BuildSpec for an Amplify App.
+   *
+   * @default - No build spec
+   */
+  readonly buildSpec?: BuildSpec;
+
+  /**
+   * Custom rewrite / redirect rules for an Amplify App.
+   *
+   * @default - No custom rules
+   */
+  readonly customRules?: CustomRule[];
+
+  /**
+   * Description for an Amplify App.
+   *
+   * @default - No description
+   */
+  readonly description?: string;
+
+  /**
+   * Environment variables map for an Amplify App.
+   *
+   * @default - No environment variables
+   */
+  readonly environmentVariables?: EnvironmentVariable[];
+
+  /**
+   * IAM Service Role
+   *
+   * @default - No service role
+   */
+  readonly serviceRole?: IRole;
+
+  /**
+   * Name for the Amplify App.
+   */
+  readonly appName: string;
+
+  /**
+   * OAuth token for 3rd party source control system for an Amplify App, used to create webhook and read-only
+   * deploy key. OAuth token is not stored.
+   *
+   * @default - No oauth token
+   */
+  readonly oauthToken?: SecretValue;
+
+  /**
+   * Repository for an Amplify App.
+   *
+   * Required to be in HTTP URL format: https://github.com/awslabs/aws-cdk
+   * If you use a Git URL, it will error.
+   */
+  readonly repository: string;
+
+  /**
+   * Tag for an Amplify App
+   *
+   * @default - No tags applied.
+   */
+  readonly tags?: Tag[];
+}
+
+/**
+ * The CustomRule property type allows you to specify redirects, rewrites, and reverse proxies.
+ * Redirects enable a web app to reroute navigation from one URL to another.
+ */
+export interface CustomRule {
+  /**
+   * The condition for a URL rewrite or redirect rule, e.g. country code.
+   *
+   * @default - No condition
+   */
+  readonly condition?: string;
+
+  /**
+   * The source pattern for a URL rewrite or redirect rule.
+   */
+  readonly source: string;
+
+  /**
+   * The status code for a URL rewrite or redirect rule.
+   *
+   * @default - No status
+   */
+  readonly status?: string;
+
+  /**
+   * The target pattern for a URL rewrite or redirect rule.
+   */
+  readonly target: string;
+}
+
+/**
+ * Custom Rules Resolver
+ */
+export class CustomRulesResolver implements IResolvable {
```

Comment:
Should not export this class.

Reply:
I kind of have no choice – I use it in other files and typescript won't allow me to use it without exporting it.